include/bout/deriv_store.hxx

```diff
@@ -0,0 +1,484 @@
+
+        //-------------------------------------------------------------
+        // Unstaggered
+        //-------------------------------------------------------------
+
+        // The direction specific section to consider
+        auto specificSection = options->getSection(direction.second);
+
+        // Find the appropriate value for theDefault either from
+        // the input file or if not found then use the value in
+        // initialDefaultMethods
+        if (specificSection->isSet(derivName)) {
+          specificSection->get(derivName, theDefault, "");
+        } else if (backupSection->isSet(derivName)) {
+          backupSection->get(derivName, theDefault, "");
+        } else {
+          theDefault = initialDefaultMethods[theDerivType];
+        }
+
+        // Now we have the default method we should store it in defaultMethods
+        theDefault = uppercase(theDefault);
+        defaultMethods[getKey(theDirection, STAGGER::None, theDerivTypeString)] =
+            theDefault;
+        output_info << "The default method for derivative type " << theDerivTypeString;
+        output_info << " in direction " << DIRECTION_STRING(theDirection);
+        output_info << " is " << theDefault << "\n";
+
+        //-------------------------------------------------------------
+        // Staggered
+        //-------------------------------------------------------------
+
+        // The direction specific section to consider with staggering
+        specificSection = options->getSection(direction.second + "stag");
+
+        // Find the appropriate value for theDefault either from
+        // the input file. Note if the specific section for staggering isn't
+        // found then we leave the default as for the non-staggered version
+        if (specificSection->isSet(derivName)) {
+          specificSection->get(derivName, theDefault, "");
+        }
+
+        // Now we have the default method we should store it in defaultMethods
+        theDefault = uppercase(theDefault);
+        defaultMethods[getKey(theDirection, STAGGER::L2C, theDerivTypeString)] =
+            theDefault;
+        defaultMethods[getKey(theDirection, STAGGER::C2L, theDerivTypeString)] =
+            theDefault;
+        output_info << "The default method for staggered derivative type "
+                    << theDerivTypeString;
+        output_info << " in direction " << DIRECTION_STRING(theDirection);
+        output_info << " is " << theDefault << "\n";
+      }
+    }
+  }
+
+  /// The following stores what actual method to use when DIFF_DEFAULT
+  /// is passed. The key is determined using the getKey routine here,
+  /// where the name we pass is determined by the type of method (standard,
+  /// upwind etc.). Note for now we'll always use STAGGER::None as we
+  /// currently assume the default method is independent of staggering --
+  /// it might be useful to relax this assumption!
+  storageType<std::size_t, std::string> defaultMethods;
```

Comment:
Should this be private?

Reply:
Yes it should! I think this was public because of a different original design.